lib/async.js

```diff
@@ -46,7 +46,7 @@ module.exports = function resolve (x, opts, cb) {
     else loadNodeModules(x, y, function (err, n, pkg) {
         if (err) cb(err)
         else if (n) cb(null, n, pkg)
-        else if (core[x]) return cb(null, x);
+        else if (core[x]) return cb(null, x, null);
```

Comment:
Hello, the problem is a rare case of libraries combination.
When multiple values are returned from a thunkified function that is run inside "co", the result is an array. See https://github.com/tj/co/blob/master/index.js#L139
Maybe it should always return an array, but it's about convenience vs. consistency.
On the other side "resolve", as a lower level lib, could handle problems like this, by always returning the same number of values.

Also your "slice(0, 2)" version inside the generator function, is not equivalent for the "unpatched" version. The first one does array destructuring, so it expects the sourced result to be an array and returns an array. The second one has the same behavior only for an array result, but for single value results it returns a string.
```javascript
const [filePath, pkg] = yield resolve('path')
return [filePath, pkg] //=> ['p', 'a']
// vs.
return (yield resolve('path')).slice(0, 2) //=> 'pa'
```
In the "patched" version we get `[ 'path', null ]` from both.